hq/app/schedule/IamDisableAccessKeys.scala

```diff
@@ -0,0 +1,53 @@
+package schedule
+
+import aws.AwsAsyncHandler.{awsToScala, handleAWSErrs}
+import aws.iam.IAMClient.SOLE_REGION
+import aws.{AwsClient, AwsClients}
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagementAsync
+import com.amazonaws.services.identitymanagement.model.{UpdateAccessKeyRequest, UpdateAccessKeyResult}
+import model._
+import play.api.Logging
+import schedule.IamFlaggedUsers.{hasOutdatedHumanKey, hasOutdatedMachineKey}
+import schedule.IamListAccessKeys.listAccountAccessKeys
+import utils.attempt.Attempt
+
+import scala.concurrent.ExecutionContext
+
+object IamDisableAccessKeys extends Logging {
+
+  def disableAccessKeys(account: AwsAccount, vulnerableUsers: Seq[VulnerableUser], iamClients: AwsClients[AmazonIdentityManagementAsync])
+    (implicit ec: ExecutionContext): Unit = {
+    // this does the work of taking our vulnerable users who have been flagged as potentially needing their access keys disabled
+    // and converts that vulnerableUser into a user that has it's access key id attached to it
+    val vulnerableUserWithAccessKeyId: Attempt[List[VulnerableUserWithAccessKeyId]] = listAccountAccessKeys(account, vulnerableUsers, iamClients)
+    vulnerableUserWithAccessKeyId.map { users =>
```

Comment:
This feels like a good place to add a little error handling - so we could change this to something like the pseudocode:
`vulnerableUserWithAccessKeyId.fold(log.warn('failed to list access keys')) { ... }`
and putting the following on the end of the for comprehension:
```
).asFuture.recover { case err =>
    logger.warn("failed to disable access keys")
}
```